2016-04-08-rules-and-queries.md

```diff
@@ -0,0 +1,342 @@
+
+* `project_id` - ID of the project. Changes in products by rules are applied to this project.
+* `rule_id` - ID of the rule in Mergado, each rule is represented by one row in the UI.
+* `apply_log_id` - ID of the rule application's log.
+* `request_id` - ID of the request different for each request. Note that the ID doesn't change when the request is retried (see Retrying on Errors bellow).
+* `current_format` - Name of the format in which the products are represented when this rule is applied.
+* `data` - A list of products.
+    + `id` - ID of the product.
+    + `created_at` - When the product has been imported in Mergado.
+    + `updated_at` - The last time the product changed.
+    + `output_changed_at` - The last time the product changed its output values.
+    + `data` - Nested structure representing product data. These values are altered by rules with higher priority (applied sooner in the chain of rules).
+    + `metadata` - Cached data for the whole application procces that you send before in response for this particular product in another rule.
+
+The request is considered to be a success if the server replies with a 200 OK HTTP status code and the body of the response contains products’ data in the same format. The application is not required to return all products, it is required to return only the products and subtrees with top level elements as roots that were processed and should be changed in some way. If application wants to change the values ​​of elements, it is necessary to send the entire subtree in the response, starting with the top level element, in its final form. For example, the server's response might be:
+
+```json
+{
+    "data": [
+        {
+            "id": "234",
+            "data": {
+                "ITEM_ID": [{"value": "11111"}],
+                "IMGURL": [{"value": "www.image4.com"}{"value": "www.image5.com"}]
+                "GIFTS": [{
+                	"attributes": {
+                		"ID":[{"value": "KVP-175"}]
+                		"DURATION":[{"value": "3 weeks"}]
+                	}, 
+                	"children": {"GIFT":[{"value": "Discount 10$"}]}
+                }]
+                ""
+            },
+            "metadata": {
+                "..."
+            }
+        }
+    ]
+}
+```
+
+#### Changing the value of an element
+It is possible to change the value of the element by changing the `value` field of the given element. In case of changing the value of a nested element, it is necessary to send the entire structure in which it is located (whole subtree from top level elements to tree leaves).
+
+#### Creation of a new element value 
+Newly created values ​​for the product element must be inserted at the end of the list of values ​​for the given element. For example `"IMGURL": [{"value": "www.image1.com"}, {"value": "www.image5.com"}, {"value": "new_value_of_IMGURL"}] ` means, that application wants to add new value (new_value_of_IMGURL) for element IMGURL.
```

Comment:
> For example `"IMGURL":

asi chybí čárka: For example, `"IMGURL":

> means, that

tady nemá být čárka (dependent clauses)

> application wants to add new value (new_value_of_IMGURL) for element IMGURL.

chybí členy u `application`, `new value` a `element`